core/commands.py

```diff
@@ -287,7 +293,7 @@ def update_sha512_sums_file(self):
 
         self.log(f"Uploading {fname}...")
         progress = MinioProgress()
-        self.minio.fput_object(self.bucket, dest_name, temp_file.name, progress=progress)
+        self.minio.fput_object(self.bucket, dest_name, temp_file.name, progress=progress, content_type="text/plain")
```

Comment:
Se a criação do SHA512SUMS for responsabilidade de outra classe, talvez possamos usar o mesmo mecanismo de upload dos arquivos para ele também (assim o `fput` ficaria centralizado, que evita corner cases).

Reply:
Confesso que não consegui vislumbrar como refatorar essa parte não =S

Mas realmente, essa comunicação com o minio me incomodou também. Tem a ver com a API dele que achei bem limitada e confusa. Do tipo, esse lance de eu ter que instanciar um `progress` antes de chamar o método, por exemplo, é algo que complexifica desnecessariamente. Eu voto por abrirmos uma issue para documentar essas questões e liberarmos esse PR dado que esse é o último ponto dos seus comentários que ficou aberto.